tools/java_grpc_library.bzl

```diff
@@ -0,0 +1,228 @@
+# This is copied verbatim from
```

Comment:
if it is possible to keep the grpc dependncy in workspace file, I'd prefer that as well

Reply:
while technically possible, that would require an even uglier workaround because grpc-java requires downloading custom dependencies with rules_jvm_external and their names are clash with the names of our dependencies

Reply:
Can something like this work? Import the grpc repository similar to the old bzlmod PR but in the Module.bazel file do mapping to the repo names used by the plugin in `use_repo`, for example (`jsr305_annotations="com_google_code_findbugs_jsr305"`). Then use [repo_mapping](https://github.com/bazelbuild/rules_jvm_external?tab=readme-ov-file#repository-remapping), in the `http_archive` of the grpc repository with the mapping from the repo names we have in the plugin code to the names used by grpc.

It should look like:

```
http_archive(
    name = "io_grpc_grpc_java",
    sha256 = "3bcf6be49fc7ab8187577a5211421258cb8e6d179f46023cc82e42e3a6188e51",
    strip_prefix = "grpc-java-1.59.0",
    url = "https://github.com/grpc/grpc-java/archive/refs/tags/v1.59.0.tar.gz",
    repo_mapping = {
    	"@com_google_code_findbugs_jsr305": "@jsr305_annotations",
    }
)
```